core/src/executor/evaluate/expr.rs

```diff
@@ -78,3 +78,16 @@ pub fn between<'a>(
 
     Ok(Evaluated::from(Value::Bool(v)))
 }
+
+pub fn array_index<'a>(obj: Evaluated<'a>, indexes: Vec<Evaluated<'a>>) -> Result<Evaluated<'a>> {
+    let value = match obj {
+        Evaluated::Value(value) => value,
+        _ => return Err(EvaluateError::MapOrListTypeRequired.into()),
+        // TODO Q. 의미도 안맞고 실제로 map, list check는 selector에서 ..
```

Comment:
Actually, this match statement does not check whether the value is either`map` or `list` type.. Better name for this error?